### PR TITLE
Add sharding-orchestration-core-new module

### PR DESCRIPTION
**Pull request by wgy8283335**
Fixes #3185 .

Changes proposed in this pull request:
- add sharding-orchestration-core-new module
-
-

### CODE REVIEW

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/RegistryCenterServiceLoader.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        super(RegistryCenter.class);
+    }
+    
+    /**
+     * Load registry center from SPI.
+     * 
+     * @param regCenterConfig registry center configuration
+     * @return registry center
+     */
+    public RegistryCenter load(final InstanceConfiguration regCenterConfig) {
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/RegistryCenterServiceLoader.java]: I suggest the param name is `config` .

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    @Getter
+    private final ConfigurationService configService;
+    
+    private final StateService stateService;
+    
+    private final ShardingOrchestrationListenerManager listenerManager;
+    
+    public ShardingOrchestrationFacade(final OrchestrationConfiguration orchestrationConfig, final Collection<String> shardingSchemaNames) {
+        InstanceConfiguration registryCenterConfiguration = getConfigurationByOrchestrationType(orchestrationConfig.getInstanceConfigurationMap(),
+                "registry_center");
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java]: I suggest `registry_center` can get with `OrchestrationType`, not string directly.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        regCenter = new RegistryCenterServiceLoader().load(registryCenterConfiguration);
+        isOverwrite = Boolean.valueOf(registryCenterConfiguration.getProperties().getProperty("isOverwrite"));
+        stateService = new StateService(registryCenterConfiguration.getNamespace(), regCenter);
+        InstanceConfiguration configCenterConfiguration = getConfigurationByOrchestrationType(orchestrationConfig.getInstanceConfigurationMap(),
+                "config_center");
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java]: The same to above.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    
+    private InstanceConfiguration getConfigurationByOrchestrationType(final Map<String, InstanceConfiguration> map, final String type) {
+        if (null == map || null == type) {
+            return null;
+        }
+        for (Entry entry: map.entrySet()) {
+            InstanceConfiguration configuration = (InstanceConfiguration) entry.getValue();
+            if (type.equals(configuration.getOrchestrationType())) {
+                return configuration;
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java]: `Entry<String, InstanceConfiguration>`  and don't need Class cast.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/FixedTimeService.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+package org.apache.shardingsphere.orchestration.temp.internal.keygen.fixture;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.orchestration.temp.internal.keygen.LeafSnowflakeKeyGenerator;
+import org.apache.shardingsphere.orchestration.temp.internal.keygen.TimeService;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+@RequiredArgsConstructor
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/FixedTimeService.java]: Need to add Class header comment.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/PreviousTimeService.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.orchestration.temp.internal.keygen.fixture;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.orchestration.temp.internal.keygen.TimeService;
+
+@RequiredArgsConstructor
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/PreviousTimeService.java]: Need to add Class header comment.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "testService1");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("distributedLockManagementType", "ForthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber << 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+        Properties properties = new Properties();
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "testService1");
+        properties.setProperty("distributedLockManagementType", "ForthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber << 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("step", "3");
+        properties.setProperty("leafKey", "test_table_5");
+        properties.setProperty("registryCenterType", "ThirdTestRegistryCenter");
+        leafSegmentKeyGenerator.setProperties(properties);
+        int taskNumber = threadNumber * 2;
+        Set<Comparable<?>> actual = new HashSet<>();
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("initialValue", "100001");
+        properties.setProperty("leafKey", "test_table_4");
+        properties.setProperty("registryCenterType", "ThirdTestRegistryCenter");
+        leafSegmentKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber * 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+        properties.setProperty("initialValue", "100001");
+        properties.setProperty("step", "3");
+        properties.setProperty("leafKey", "test_table_2");
+        properties.setProperty("registryCenterType", "ThirdTestRegistryCenter");
+        leafSegmentKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber * 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java [hunk 12]
```diff
@@ -0,0 +1,334 @@
...
+        properties.setProperty("step", "3");
+        properties.setProperty("digest", "name:123456");
+        properties.setProperty("leafKey", "test_table_3");
+        properties.setProperty("registryCenterType", "ThirdTestRegistryCenter");
+        leafSegmentKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber * 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: I think this blank line should remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/configcenter/ConfigCenterServiceLoader.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        super(ConfigCenter.class);
+    }
+    
+    /**
+     * Load config center from SPI.
+     * 
+     * @param configCenterConfig config center configuration
+     * @return config center
+     */
+    public ConfigCenter load(final InstanceConfiguration configCenterConfig) {
```
[the remark this answers is not in the document]
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/configcenter/ConfigCenterServiceLoader.java]: THX. It's a good idea.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGenerator.java [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+                tryPutCacheId(newId);
+            }
+        });
+    }
+    
+    private InstanceConfiguration getInstanceConfiguration() {
+        InstanceConfiguration result = new InstanceConfiguration(getRegistryCenterType(), properties);
+        result.setNamespace(DEFAULT_NAMESPACE);
+        result.setServerLists(getServerList());
+        //result.setDigest(getDigest());
```
[the remark this answers is not in the document]
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGenerator.java]: THX. I‘ve got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/RegistryCenterServiceLoader.java (hunk 1 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/RegistryCenterServiceLoader.java]: THX. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java (hunk 2 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java]: OK. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java (hunk 4 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java]: THX. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/FixedTimeService.java (hunk 5 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/FixedTimeService.java]: THX. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/PreviousTimeService.java (hunk 6 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/fixture/PreviousTimeService.java]: THX. I've modified it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java (hunk 7 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: THX. I have resolved it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 11 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: THX. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java (hunk 8 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: THX. I've resolved it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 9 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: THX. I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 10 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: THX. I have resolved it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 12 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: THX.I've got it.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java (hunk 7 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: I‘m sorry, I make a mistake. This blank line should be remained and don't need remove.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java (hunk 8 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: The save to above.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 9 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: The save to above.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 10 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: The save to above.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 11 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: The save to above.

File: sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java (hunk 12 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSegmentKeyGeneratorTest.java]: The save to above.

**Pull request merged**